notebooks/Inference_from_SavedModel.ipynb

```diff
@@ -0,0 +1,479 @@
+{
```

Comment:
<p>Why the ground truth is pixelated like this? </p><p> </p><p>This wasn't the case for https://github.com/deep-diver/segformer-tf-transformers. </p><p> </p><p>Also, you might want to add a note on why the predictions aren't good so that the readers are aware. </p><p> </p><p>Also, are we not exporting the <code>SavedModel</code> to accept bytestrings like we had discussed?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-diver/semantic-segmentation-ml-pipeline/pull/15/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
> Also, are we not exporting the SavedModel to accept bytestrings like we had discussed?

we did, but it seems like the custom signature only works when we serve the model in TF Serving? 

> maybe there is some problem when reducing the resolution? I am using the exactly the same logic to https://github.com/deep-diver/segformer-tf-transformers for visualization

Reply:
> we did, but it seems like the custom signature only works when we serve the model in TF Serving?

I don't think so. Since our dual deployments project used that logic seamlessly. So not sure. 

> maybe there is some problem when reducing the resolution? I am using the exactly the same logic to https://github.com/deep-diver/segformer-tf-transformers for visualization

Spot on! Let's add a note about it.

Reply:
when calling `predict()` method, it calls the registered signature?

Reply:
Oh for that signature either needs to be called with Vertex AI deployed endpoint or via concrete functions.

Reply:
oh so, I will update the notebook by adding additional section to test prediction on Vertex AI endpoint then :)

Reply:
The model shown in this notebook was derived through Vertex pipeline?

Reply:
yeah the model is just downloaded from the output artifact of Vertex AI(Trainer component), and I just load it up locally without any serving framework. So, I just called `predict()` method directly

Reply:
You were totally right in doing that! Thanks for clarifying! You can follow the endpoint prediction code we developed here: 

https://github.com/sayakpaul/deploy-hf-tf-vision-models/blob/main/hf_vision_model_vertex_ai/test-vertex-ai-endpoint.ipynb

Reply:
will let you know when adding Vertex AI endpoint to the notebook! Also thanks for sharing the reference notebook :)